### PR TITLE
MC-38105: Unable to set native session handler that is different from the one define in php.ini

### LINKED ISSUE #24717: Unable to set native session handler that is different from the one define in php.ini after update to Magento 2.3
### Preconditions (*)

2.4-develop

### Steps to reproduce (*)

1. Configure env.php to use memcached as save_handler as[ instructed in the documentation](https://devdocs.magento.com/guides/v2.3/config-guide/memcache/memcache_magento.html). 

### Expected result (*)

1. SessionManager should be setting memcached as used save handler in initOptions function. 

### Actual result (*)

1. SessionManager is not setting the memcached as save_handler, but is instead using the files session_handler. The save_path is set though, which result error message below:
> main.CRITICAL: Warning: SessionHandler::read(): open(127.0.0.1:11211/sess_0imeeaqmnvemdg4e3h57tat0ik, O_RDWR) failed: No such file or directory (2) in../vendor/magento/framework/Session/SaveHandler/Native.php on line 22 {"exception":"[object] (Exception(code: 0): Warning: SessionHandler::read(): open(127.0.0.1:11211/sess_0imeeaqmnvemdg4e3h57tat0ik, O_RDWR) failed: No such file or directory (2) in ../vendor/magento/framework/Session/SaveHandler/Native.php on line 22 at ../vendor/magento/framework/App/ErrorHandler.php:61)"} []

### Own foundings
New changes made in 2.3 for [SessionManager](https://github.com/magento/magento2/blob/2.3.0/lib/internal/Magento/Framework/Session/SessionManager.php#L605) are preventing the save_handler to be set as anything else than the one configured to php.ini for Native session handler.

### PR DESCRIPTION
**Pull request by engcom-Echo**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#24717

### Manual testing scenarios (*)

See: magento/magento2#24717

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/Framework/Session/SessionManagerTest.php [hunk 1]
```diff
@@ -313,22 +320,35 @@ public function testConstructor()
...
-            $this->assertEquals('db', $sessionConfig->getOption('session.save_handler'));
+            $this->assertEquals($saveMethod, $sessionConfig->getOption('session.save_handler'));
             $this->assertEquals('private_no_expire', $sessionConfig->getOption('session.cache_limiter'));
             $this->assertEquals('explicit_save_path', $sessionConfig->getOption('session.save_path'));
             $this->assertArrayHasKey('session.use_only_cookies', self::$isIniSetInvoked);
             $this->assertEquals('1', self::$isIniSetInvoked['session.use_only_cookies']);
             foreach ($sessionConfig->getOptions() as $option => $value) {
-                if ($option=='session.save_handler') {
-                    $this->assertArrayNotHasKey('session.save_handler', self::$isIniSetInvoked);
+                if ($option == 'session.save_handler' && $value == 'user') {
```
Comment (ihor-sviziev) [inline dev/tests/integration/testsuite/Magento/Framework/Session/SessionManagerTest.php]:
```suggestion
                if ($option === 'session.save_handler' && $value === 'user') {
```

**Comment by ihor-sviziev** [comment]
@magento run WebAPI Tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
For now the only one improvement is needed.

Also tests were failing, let's check results again after restart

**Comment by engcom-Echo** [comment]
> For now the only one improvement is needed.
> 
> Also tests were failing, let's check results again after restart

Thanks. I think that the failure of the tests will be repeated and we need to change the solution.

**Comment by ihor-sviziev** [comment]
@engcom-Echo pls just merge latest changes from 2.4-develop. I believe these failures related to recent changes

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8460](https://jira.corp.magento.com/browse/ENGCOM-8460) has been created to process this Pull Request

**Review by viktorpetryk — APPROVED** [review]

**Comment by engcom-Kilo** [comment]
@magento import pull request to https://github.com/magento-tsg/magento2ce

**Review by engcom-Kilo — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
@engcom-Kilo the pull request successfully imported.

**Comment by engcom-Oscar** [comment]
:heavy_check_mark: QA passed
Was checked case from Manual testing scenarios 
:x: **Before:**
main.CRITICAL: Warning: SessionHandler::read(): open(127.0.0.1:11211/sess_0imeeaqmnvemdg4e3h57tat0ik, O_RDWR) failed: No such file or directory (2) in../vendor/magento/framework/Session/SaveHandler/Native.php on line 22 {"exception":"[object] (Exception(code: 0): Warning: SessionHandler::read(): open(127.0.0.1:11211/sess_0imeeaqmnvemdg4e3h57tat0ik, O_RDWR) failed: No such file or directory (2) in ../vendor/magento/framework/Session/SaveHandler/Native.php on line 22 at ../vendor/magento/framework/App/ErrorHandler.php:61)"} []
:heavy_check_mark: **After:**
No errors on homepage and in Admin on backend

**Pull request merged**